font/src/ft/mod.rs

```diff
@@ -253,123 +246,107 @@ impl FreeTypeRasterizer {
         let primary_font = pattern.render_prepare(config, primary_font);
 
         // Hash pattern together with request pattern to include requested font size in the hash
-        let primary_font_id = FontID::new(hash, primary_font.hash());
-
-        // Reload already loaded faces and drop their fallback faces
-        let font_key = if let Some(font_key) = self.keys.remove(&primary_font_id) {
-            let fallback_list = self.fallback_lists.remove(&font_key).unwrap_or_default();
-
-            for fallback_font in &fallback_list.list {
-                if let Some(ff_key) = self.keys.get(&fallback_font.id) {
-                    // Skip primary fonts, since these are all reloaded later
-                    if !self.fallback_lists.contains_key(&ff_key) {
-                        self.faces.remove(ff_key);
-                        self.keys.remove(&fallback_font.id);
-                    }
-                }
-            }
+        let primary_font_key = FontKey::from_pattern_hashes(hash, primary_font.hash());
 
-            let _ = self.faces.remove(&font_key);
-            Some(font_key)
-        } else {
-            None
-        };
+        // Return if we already have the same primary font
+        if self.fallback_lists.contains_key(&primary_font_key) {
+            return Ok(primary_font_key);
+        }
 
-        // Reuse the font_key, since changing it can break library users
-        let font_key = self
-            .face_from_pattern(&primary_font, primary_font_id, font_key)
-            .and_then(|pattern| pattern.ok_or_else(|| Error::MissingFont(desc.to_owned())))?;
+        // Load font if we haven't loaded one
```

Comment:
```suggestion
        // Load font if we haven't loaded it yet
```

Since `one` is not ideal here, we only ever want to load one when talking about primary fonts.